src/background/sidebar-injector.js

```diff
@@ -27,6 +27,10 @@ function setClientConfig(config) {
   document.head.appendChild(script);
 }
 
+function hypothesisConfigExists() {
+  return !!document.querySelector('script.js-hypothesis-config[data-remove-on-unload]');
```

Comment:
I have tried to make the selector as specific as possible, but maybe `script.js-hypothesis-config` would be enough.